packages/api-server/src/methods/normalizer.ts

```diff
@@ -30,10 +30,20 @@ export class EthNormalizer {
     // otherwise read operation might fail the balance check.
     const gasPrice = txCallObj.gasPrice || "0x0";
 
-    // TODO: set default gas limit to min(maxBlockGas, userBalanceAvailableGas)
+    // set default gas limit to min(maxBlockGas, userBalanceAvailableGas)
+    // TODO: use real blockAvailableGas to replace POLY_MAX_BLOCK_GAS_LIMIT
     const maxBlockGasLimit =
       "0x" + BigInt(POLY_MAX_BLOCK_GAS_LIMIT).toString(16);
-    const gas = txCallObj.gas || maxBlockGasLimit;
+    let gas = txCallObj.gas;
+    if (gas == null) {
+      gas =
+        +gasPrice === 0
+          ? maxBlockGasLimit
+          : min(
```

Comment:
This min can't handle `BigInt`

Reply:
there is no `BigInt` type, all use `HexNumber`

Reply:
oh wait, you mean the overflow problem? I will update the min function

Reply:
updated in base pr https://github.com/nervosnetwork/godwoken-web3/pull/519/commits/ac4a99e9761110c509e63bcc44ccc8d4b9d0fce8